src/core/server/logging/README.md

```diff
@@ -264,6 +264,127 @@ The maximum number of files to keep. Once this number is reached, oldest files w
+modifying the provided log events before passing them along to another
+appender.
+
+```yaml
+logging:
+  appenders:
+    my-rewrite-appender:
+      type: rewrite
+      appenders: [console, file] # name of "destination" appender(s)
+      policy:
+        # ...
+```
+
+The most common use case for the `RewriteAppender` is when you want to
+filter or censor sensitive data that may be contained in a log entry.
+In fact, with a default configuration, Kibana will automatically redact
+any `authorization`, `cookie`, or `set-cookie` headers when logging http
+requests & responses.
+
+To configure additional rewrite rules, you'll need to specify a `RewritePolicy`.
+
+#### Rewrite Policies
+
+Rewrite policies exist to indicate which parts of a log record can be
+modified within the rewrite appender.
+
+**Meta**
+
+The `meta` rewrite policy can read and modify any data contained in the
+`LogMeta` before passing it along to a destination appender.
+
+Meta policies must specify one of three modes, which indicate which action
+to perform on the configured properties:
+- `add` creates a new property at the provided `path`, skipping properties which already exist.
+- `update` updates an existing property at the provided `path` without creating new properties.
+- `remove` removes an existing property at the provided `path`.
+
+The `properties` are listed as a `path` and `value` pair, where `path` is
+the dot-delimited path to the target property in the `LogMeta` object, and
+`value` is the value to add or update in that target property. When using
+the `remove` mode, a `value` is not necessary.
+
+Here's an example of how you would replace any `cookie` header values with `[REDACTED]`:
+
+```yaml
+logging:
+  appenders:
+    my-rewrite-appender:
+      type: rewrite
+      appenders: [console]
+      policy:
+        type: meta # indicates that we want to rewrite the LogMeta
+        mode: update # will update an existing property only
+        properties:
+          - path: "http.request.headers.cookie" # path to property
+            value: "[REDACTED]" # value to replace at path
+```
+
+Rewrite appenders can even be passed to other rewrite appenders to apply
+multiple filter policies/modes, as long as it doesn't create a circular
+reference. Each rewrite appender is applied sequentially (one after the other).
+```yaml
+logging:
+  appenders:
+    remove-stuff:
+      type: rewrite
+      appenders: [add-stuff] # redirect to the next rewrite appender
+      policy:
+        type: meta
+        mode: remove
+        properties:
+          - path: "http.request.headers.authorization"
+          - path: "http.request.headers.cookie"
+          - path: "http.request.headers.set-cookie"
+    add-stuff:
+      type: rewrite
+      appenders: [console] # output to console
+      policy:
+        type: meta
+        mode: add
+        properties:
+          - path: "hello"
+            value: "world" # creates { hello: 'world' } at the LogMeta root
```

Comment:
is the `add` mode without any value interpolation really relevant?

Reply:
Fair question, I mostly included an `add` mode as [log4j has one](https://logging.apache.org/log4j/2.x/manual/appenders.html#RewritePolicy), and it didn't really require extra work to support.

But admittedly update/remove seem to be the more useful modes. (FWIW the example usage log4j gives for `add` is injecting a static product key). Not opposed to removing it though, if we don't feel it's worth keeping. Thoughts?

Reply:
If we don't have a need to support an `add` mode, maybe we don't need it right now.
My vote is that we don't have a direct need for it and shouldn't include it right now.

Reply:
Removed